torch/csrc/jit/mobile/backport_manager.cpp

```diff
@@ -0,0 +1,238 @@
+      auto data = std::get<0>(data_ptr).get();
+      auto size = std::get<1>(data_ptr);
+      writer.writeRecord(record, data, size);
+    }
+  }
+}
+
+bool check_bytecode_version(
+    const std::vector<c10::IValue>& bytecode_values,
+    const int64_t expect_bytecode_version) {
+  if (bytecode_values.empty()) {
+    TORCH_WARN("Empty bytecode archive.");
+    return false;
+  } else if (bytecode_values[0] != expect_bytecode_version) {
+    TORCH_WARN(
+        "Expect bytecode version ",
+        expect_bytecode_version,
+        ", but it gets ",
+        bytecode_values[0]);
+    return false;
+  }
+  return true;
+}
+
+} // namespace
+
+// To add next backport
+// function, for example, backport_vn_to_vn-1, create an anonymous namespace
+// with a backport_vn_to_vn-1 function + other necessary customized function. If
+// a function can be reused by other backport functions, move it to the utility
+// function group. It will be easier to split out backport_manager.cpp to
+// smaller files when it grows too long.
+
+// The functions needed for backport model from v5 to v4.
+namespace {
+
+void writeArchiveV4(
+    PyTorchStreamWriter& writer,
+    const std::string& archive_name,
+    const c10::IValue& value) {
+  std::vector<char> data;
+
+  // Vector to capture the run-time class types during pickling the IValues
+  std::vector<c10::ClassTypePtr> memoizedClassTypes;
+  Pickler data_pickle(
+      [&](const char* buf, size_t size) {
+        data.insert(data.end(), buf, buf + size);
+      },
+      nullptr,
+      nullptr,
+      &memoizedClassTypes);
+  data_pickle.protocol();
+  data_pickle.pushIValue(value);
+  data_pickle.stop();
+  size_t i = 0;
+  std::string prefix = archive_name + "/";
+
+  for (const auto& td : data_pickle.tensorData()) {
+    WriteableTensorData writable_td = getWriteableTensorData(td);
+    std::string fname = prefix + c10::to_string(i++);
+    writer.writeRecord(fname, writable_td.data(), writable_td.sizeInBytes());
+  }
+  std::string fname = archive_name + ".pkl";
+  writer.writeRecord(fname, data.data(), data.size());
+}
+
+bool backport_v5_to_v4(
+    PyTorchStreamReader& reader,
+    PyTorchStreamWriter& writer) {
+  // 1) read from archive `bytecode` archive
+  std::vector<IValue> bytecode_values = get_bytecode_values(reader);
+  if (!check_bytecode_version(bytecode_values, kBytecodeVersionV5)) {
+    TORCH_WARN("Incorrect bytecode version for input model.");
+    return false;
+  }
+
+  // 2) Copy everything except bytecode related to new output
+  copy_non_bytecode(reader, writer);
+
+  // 3) write `bytecode` archive
+  // Update the bytecode version in bytecode.pkl
+  update_bytecode_version(bytecode_values, kBytecodeVersionV4);
+  // Construct the list of ivalues to a big tuple
+  auto bytecode_tuple = c10::ivalue::Tuple::create(std::move(bytecode_values));
+  // write `bytecode` archive
+  writeArchiveV4(writer, "bytecode", bytecode_tuple);
+  return true;
+}
+} // namespace
+
```

Comment:
Move to the bottom so the file grows with implementations being appended.

Reply:
I try and it's not working..after looking into it, looks like if it's in anonymous namespace, forward declare doesn't work. There are a couple of options:
1. don't use anonymous namespace for backport_v5_to_v4, and forward declare. I think it's still good to keep the anonymous namesapce for each backport_vn_to_vn-1 function, because the dependency is clearer.
2. Move these line of code to the bottom, after all backport_vn_to_vn-1 functions. It doesn't look nice and it's far from other BackportManager implementation.
```
BackportManager::BackportManager() {
  registerBytecodeBackportFunction(kBytecodeVersionV5, backport_v5_to_v4);
}
```
I feel like the current way is still better, despite the file will expend in the middle, but not at the end of file.